src/tests/ftest/cart/iv_client.c

```diff
@@ -358,8 +392,90 @@ test_iv_update(struct iv_key_struct *key, char *str_value, bool value_is_hex,
 		DBG_PRINT("Update FAILED; rc = %ld\n", output->rc);
 
 	rc = crt_req_decref(rpc_req);
+exit_code:
 	assert(rc == 0);
+	return 0;
+}
 
+static int
+test_iv_set_grp_version(char *arg_version, char *arg_timing)
+{
+	struct RPC_SET_GRP_VERSION_in		*input;
+	struct RPC_SET_GRP_VERSION_out		*output;
+	crt_rpc_t				*rpc_req;
+	int					 rc;
+	int					 version;
+	int					 time = 0;
+	char					*tmp;
+
+	/* See if string contais valid hex characters.  If so assume
+	 * base 16.  Else, default to base 10.
+	 */
+	tmp = strpbrk(arg_version, "abcdABCDxX");
+	if (tmp == NULL)
+		version = strtol(arg_version, NULL, 10);
+	else
+		version = strtol(arg_version, NULL, 16);
+
+	DBG_PRINT("Attempting to set group version to 0x%08x: %d\n",
+		 version,  version);
+
+	/* decode timing for changing version */
+	if (arg_timing != NULL) {
+		/* Avoid check patch warning */
+		time = strtol(arg_timing, NULL, 10);
+	}
+
+	prepare_rpc_request(g_crt_ctx, RPC_SET_GRP_VERSION, &g_server_ep,
+			    (void **)&input, &rpc_req);
+
+	/* Fill in the input structure */
+	input->version = version;
+	input->timing = time;
+
+	/* send the request */
+	send_rpc_request(g_crt_ctx, rpc_req, (void **)&output);
+
+	/* Check of valid output */
+	if (output->rc == 0)
+		DBG_PRINT("Grp Set Version PASSED 0x%x : %d\n",
+			 version, version);
+	else
+		DBG_PRINT("Grp Set Version FAILED 0x%x : %d\n",
+			 version, version);
+
+	rc = crt_req_decref(rpc_req);
+
+	assert(rc == 0);
+	return 0;
+}
+
+static int
+test_iv_get_grp_version()
+{
+	struct RPC_GET_GRP_VERSION_in		*input;
+	struct RPC_GET_GRP_VERSION_out		*output;
+	crt_rpc_t				*rpc_req;
+	int					 rc;
+	int					 version = 0;
+
+	prepare_rpc_request(g_crt_ctx, RPC_GET_GRP_VERSION, &g_server_ep,
+			    (void **)&input, &rpc_req);
+
+	DBG_PRINT("Attempting to get group version\n");
+	send_rpc_request(g_crt_ctx, rpc_req, (void **)&output);
+
+	/* Check for valid output */
+	version = output->version;
+	if (output->rc != 0)
+		DBG_PRINT("Grp Get Version FAILED 0x%08x : %d\n",
```

Comment:
might be useful here to print output->rc instead of version as we cant rely on group version when rc != 0